### PR TITLE
Fix bug when multiplying SparsePauliOp by np.int64

### LINKED ISSUE #5408: SparsePauliOp.__getitem__ causes a problematic behavior with numpy’s __mul__
### Information

- **Qiskit Terra version**:master (b5d5a3b)
- **Python version**: 3.8.5
- **Operating system**: macOS

### What is the current behavior?

SparsePauliOp does not support a multiplication with np.int64, np.float64, and...

### Steps to reproduce the problem

For example,
```python
from qiskit.quantum_info import SparsePauliOp, Pauli
import  numpy as  np
print(np.int(2) * SparsePauliOp(Pauli(label='X')))
print('--------')
print(np.int64(2) * SparsePauliOp(Pauli(label='X')))
```
outputs
```
SparsePauliOp([[ True, False]],
              coeffs=[2.+0.j])
--------
[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[SparsePauliOp([[ True, False]],
              coeffs=[2.+0.j])]]]]]]]]]]]]]]]]]]]]]]]]]]]]]]]]
```

### Suggested solutions

There are many solutions.
One is `__getitem__` returns a tuple of Pauli and coeff.
Current SprsePauliOp can get an item again and again. e.g. sp is instance of SparsePauliOp and we can get `sp[0][0][0][0]...`

I found this in https://github.com/Qiskit/qiskit-terra/pull/5261

### PR DESCRIPTION
**Pull request by chriseclectic**
### Summary

Fixes #5408

### Details and comments

Fix is to add dispatch mechanism for numpy ufuncs to BaseOperator which maps `np.multiply` to the `BaseOperator._multiply` method.

### CODE REVIEW

**Review by ikkoham — APPROVED** [review]
LGTM. I have checked that the issue is fixed by this PR.
Moreover, 
```python
np.array([3, 2])*SparsePauliOp(Pauli(label="XX"))
```
raises `QiskitError: 'other is not a number'` correctly.

File: qiskit/quantum_info/operators/base_operator.py [hunk 1]
```diff
@@ -636,3 +636,10 @@ def __sub__(self, other):
 
     def __neg__(self):
         return self._multiply(-1)
+
+    def __array_ufunc__(self, ufunc, method, *args, **kwargs):
+        # Dispatch numpy multiply ufunc to multiple
+        if (ufunc == np.multiply and method == '__call__') and not kwargs:
```
Comment (mtreinish) [inline qiskit/quantum_info/operators/base_operator.py:642]: Are there any other cases where numpy gets confused, or is it just calls to `np.multiply`? The list of universal functions is pretty long https://numpy.org/doc/stable/reference/ufuncs.html#available-ufuncs

**Review by mtreinish — COMMENTED** [review]
LGTM, just one question inline. I also wonder if this is just hacking around a specific instance of something else getting confused and they'll keep cropping up until we figure out the root cause.

**Comment by chriseclectic** [comment]
After further testing I think this can also be solved by just adding `__array_priority__ = x` for x > 10 to the `BaseOperator` class.

File: releasenotes/notes/fix-op-ufunc-multiple-d1c59b236024f483.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+fixes:
+  - |
+    Fixes bug in :class:`~qiskit.quantum_info.SparsePauliOp` where multiplying
+    by a certain non Python builtin Numpy scalar types returned incorrect values.
```
Comment (mtreinish) [inline releasenotes/notes/fix-op-ufunc-multiple-d1c59b236024f483.yaml:5]:
```suggestion
    by a certain non Python builtin Numpy scalar types returned incorrect values.
    Fixes `#5408 <https://github.com/Qiskit/qiskit-terra/issues/5408>`__
```

**Review by mtreinish — APPROVED** [review]

**Pull request merged**